### PR TITLE
feat(sourcegraph-frontend): configure external postgres instances and update image

### PR DESCRIPTION
**Pull request by alecjacobs5401**
* Updates the `sourcegraph-frontend` to use the [custom built image that includes chamber](https://github.com/segmentio/images/pull/522)
* Updates necessary environment variables to allow [external postgres integration](https://docs.sourcegraph.com/admin/external_services/postgres#kubernetes)

### CODE REVIEW

**Comment by asaf-erlich** [comment]
What's the test plan before / after you merge this?

**Comment by alecjacobs5401** [comment]
@asaf-erlich Good question - this was going to be my set of steps:

1. Visit dev sourcegraph and make some queries/save some things
2.  Do postgres dumps/loads to migrate data to RDS
2. Update proper slack channels to let them know of "downtime" for the staging sourcegraph
3. Update deployment with this version
4. Visit dev sourcegraph and ensure all is looking good (and monitor the AWS page for each RDS instance to confirm there are connections happening)
5. Merge the [image PR](https://github.com/segmentio/images/pull/522)
6. Update this PR w/ non branched image
7. Merge/apply
8. Follow on PR (or can add to this) to remove the baked in Postgres deployment files

**Comment by asaf-erlich** [comment]
So I looked through the [sourcegraph runbook](https://paper.dropbox.com/doc/Sourcegraph-Runbook--BWVTaCAtcYKrX~kRd_J51W5wAg-5KJbmq8t7776wj5ERsCbE) to see if we have any testing suggestions and don't see much. I also don't see much testing steps in the sourcegraph public docs, mostly just upgrade details:
* https://docs.sourcegraph.com/admin/updates
* https://docs.sourcegraph.com/admin/updates/kubernetes
* https://docs.sourcegraph.com/admin/install/kubernetes/update

Maybe it wouldn't hurt to just create another slack message to them asking "is there a set of documented common tests one can manually (or automatically) execute after an upgrade to confirm basic functionality is working as expected?"

That's the only thing I might add to your plan

By the way I might look at https://docs.sourcegraph.com/admin/install/kubernetes/update a bit more, especially the tips for the blue green deployment. Not for us to do it but it mentions taking snapshots of the db and I'm wondering if we should use that over the dump procedure? Not sure what they would recommend. I don't remember if you asked them about this or not...

**Review by asaf-erlich — APPROVED** [review]
LGTM assuming you get all the details squared away on the testing procedure and do your best to document this as much as possible as you do it.

**Comment by alecjacobs5401** [comment]
@asaf-erlich Interesting find there with the snapshotting...

I think the blue-green deployment might work when we move to `tooling-3`, but would require _a lot_ of upfront work to get right within `tooling-1`.

> Not for us to do it but it mentions taking snapshots of the db and I'm wondering if we should use that over the dump procedure?

Unfortunately it looks like that assumes we are still running postgres in K8s as part of their deployment. The snapshot as well is just doing a [pg_dump](https://github.com/sourcegraph/sourcegraph-server-gen/blob/master/cmd/sourcegraph-server-gen/snapshot.go#L76) and [load](https://github.com/sourcegraph/sourcegraph-server-gen/blob/master/cmd/sourcegraph-server-gen/snapshot.go#L152) as well. It does help automate it, though, but the expectation there on loading it in the `pgsql` deployment kind of is a killer.

We might be able to use it to migrate the redis cache though if we don't want that to be busted

**Comment by asaf-erlich** [comment]
Good to know it's just doing a pg dump. Good job digging deep into more details there. I would say for the blue/green let's table that for a later discussion. I think it would require us to do a deeper investigation of what our SLOs are, what are the error budgets we have to work with, is it worth the complexity for the increased availability, etc. I suspect it's a nice-to-have for now even though I'm a fan of it.

**Pull request merged**
